src/main/functions/DidSiopMetadata.ts

```diff
@@ -30,29 +32,50 @@ function getIntersection<T>(rpMetadata: Array<T> | T, opMetadata: Array<T> | T):
   return arrayA.filter((value) => arrayB.includes(value));
 }
 
-function verifySubjectIdentifiers(subjectIdentifiers: SubjectIdentifierType | SubjectIdentifierType[]): boolean {
-  if (subjectIdentifiers || subjectIdentifiers.length) {
-    if (Array.isArray(subjectIdentifiers)) {
-      return subjectIdentifiers.includes(SIOP.SubjectIdentifierType.DID);
+function verifySubjectIdentifiers(subjectSyntaxTypesSupported: string[]): boolean {
+  if (subjectSyntaxTypesSupported.length) {
+    if (Array.isArray(subjectSyntaxTypesSupported)) {
+      return subjectSyntaxTypesSupported.includes(SIOP.SubjectIdentifierType.DID);
     }
-    return subjectIdentifiers === SIOP.SubjectIdentifierType.DID;
-  } else {
-    return false;
   }
+  return false;
 }
 
 function supportedDidMethods(rpMethods: string[] | string, opMethods: string[] | string): Array<string> {
   const supportedDidMethods = getIntersection(rpMethods, opMethods);
-  if (!supportedDidMethods.length) {
+  if (!supportedDidMethods.length || (supportedDidMethods.length === 1 && supportedDidMethods[0] === SubjectIdentifierType.DID)) {
     throw Error(SIOPErrors.DID_METHODS_NOT_SUPORTED);
   }
   return supportedDidMethods;
 }
 
-function supportedCredentialsFormats(rpCredentials: string[] | string, opCredentials: string[] | string): Array<string> {
-  const supportedCredentials = getIntersection(rpCredentials, opCredentials);
+export function supportedCredentialsFormats(rpCredentials: Format, opCredentials: Format): Format {
+  if (!rpCredentials || !opCredentials || !Object.keys(rpCredentials).length || !Object.keys(opCredentials).length) {
```

Comment:
Hmzz, not sure if this is correct. If the rp doesn't provide formats (we need to rename rpCredentials and opCredentials BTW), they are signaling they support everything right? Or at least that you probably will only get an error later down the line. Similar for the OP. If they do not provide the info I would assume they are signaling to support all kinds of formats.

Reply:
this is a little bit tricky, `vp_format` is a required object

Reply:
`vp_formats: REQUIRED. An object defining the formats, proof types and algorithms of verifiable presentations and verifiable credentials that a RP supports. Valid values are defined in the table in Section 6.7.3. of [[OpenID.VCI](https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#OpenID.VCI)] and include jwt_vc, ldp_vc, jwt_jp and ldp_vp. Formats identifiers not in the table may be supported.`

Reply:
so I don't think it can be empty

Reply:
Agree, it needs to be explicit